pontoon/sync/changeset.py

```diff
@@ -338,3 +367,53 @@ def bulk_update_translations(self):
                 'fuzzy',
                 'extra'
             ])
+            self.bulk_create_failing_checks(
+                Translation.objects.filter(pk__in=[t.pk for t in self.translations_to_update]).prefetch_related('entity')
+            )
+
+    def check_translation(self, resource, translations):
+        """
+        Return a list of failing checks. Empty list means that everything is okay with that translation.
+        """
+        CompareEntity = namedtuple(
+            'CompareEntity',
+            (
```

Comment:
.key and .val should be the only properties used in the checker.

.entry is used in the fluent checker, but it needs an actual Fluent AST node, not some string.

I suggest to set the locale on CompareFile, in case we start doing locale-dependent checks.